### DISCUSSION TITLE
chromium: How to open (in read-only mode) some html files in my "~/Documents" directory?

### DISCUSSION THREAD

**Discussion by MikeNavy** [netblue30/firejail#6091] [Q&A]
Hi,
I use chromium with Firejail 0.9.72-2 on Linux Mint 20.3 (based on Ubuntu 20.04).

I understand that, by default, access to ~/Documents is restricted.

However, I have some html files (captured with HTTRAQT, or that I have created) that are in various subdirectories of "~/Documents".

I have tried to launch chromium with several options "firejail --whitelist=/home/usrname/Documents chromium", "firejail --noblacklist=/home/username/Documents chromium", "firejail --read-only=/home/usernam/Documents chromium". I have also  copied "chromium.profile" in "/home/username/.config/firejail/" and added "whitelist=/home/username/Documents"...
All this without effect, I cannot read the html files in my Documents directory with firejailed Chromium.

What to do?

Regards,

MN

**Comment by rusty-snake** [comment]
```
noblacklist ${DOCUMENTS}
whitelist ${HOME}/Documents/my_html_files
read-only ${DOCUMENTS}
```

----

Regarding formatting, place your command in inline code (`` some `command` text ``) to prevent the `~` from being interpreted as strikethrough.

**Comment by MikeNavy** [comment]
> ```
> noblacklist ${DOCUMENTS}
> whitelist ${HOME}/Documents/my_html_files
> read-only ${DOCUMENTS}
> ```
> 
Hi, 

I have tested the three solutions, modifying "chromium.profile".
With "noblacklist..." and "read-only...", the file I want to open is not found. With "whitelist..." chromium doesn't launch.

**Reply by MikeNavy** [reply]
Hi,

I have also tested these commands with "chromium-common" profile; also tested with Firefox, with no more success. 

It works with other applications (firejail --whitelist=/home/username/documents vlc) allows opening audio or video files in Documents, but it doesn't seem to work with browsers.

I have found a workaround, I downloaded Quartz browser AppImage, and I have associated htm and html files with Quartz AppImage.

But there is something wrong with the way browsers are firejailed: access to Documents directory, at least in read-only mode, should be possible.

Regards,

MN
